src/Extensions.Hosting.AsyncInitialization/DependencyInjection/AsyncInitializationServiceCollectionExtensions.cs

```diff
@@ -81,10 +82,9 @@ public static IServiceCollection AddAsyncInitializer(this IServiceCollection ser
         /// <returns>A reference to this instance after the operation has completed.</returns>
         public static IServiceCollection AddAsyncInitializer(this IServiceCollection services, Type initializerType)
         {
-            if (initializerType == null)
-                throw new ArgumentNullException(nameof(initializerType));
-
-            return services
+            return initializerType == null
+                ? throw new ArgumentNullException(nameof(initializerType))
+                : services
```

Comment:
Please undo this change, it makes the code less readable.
(and it's not related to the feature you're implementing anyway)